reports-ui/src/main/java/io/jmix/reportsui/screen/report/edit/ReportEditor.java

```diff
@@ -161,6 +160,33 @@ protected void initNewItem(InitEntityEvent<Report> event) {
         }
     }
 
+    @Subscribe
+    public void onBeforeCommitChanges(BeforeCommitChangesEvent event) {
+        Report report = getEditedEntity();
+
+        // todo maybe generify this code
```

Comment:
May be generified, but not for Entity,  only as objects
I've done this before and it was no better than this one